packages/shared-ux/avatar/user_profile/impl/user_profiles_selectable.tsx

```diff
@@ -0,0 +1,317 @@
+              return false;
+            }
+            if (selectedOptions && selectedOptions.find((p) => p.uid === option.key)) {
+              return false;
+            }
+            return true;
+          })
+          .map((option) => option.data);
+
+        // Add all options from `props.selectedOptions` unless they have been deselected in `nextOptions`
+        if (selectedOptions && !singleSelection) {
+          selectedOptions.forEach((profile) => {
+            const match = nextOptions.find((o) => o.key === profile.uid);
+            if (!match || match.checked === 'on') {
+              values.push(profile);
+            }
+          });
+        }
+
+        onChange(values);
+      }}
+      style={{ maxHeight: height }}
+      singleSelection={singleSelection}
+      searchable
+      searchProps={{
+        placeholder:
+          searchPlaceholder ??
+          i18n.translate(
+            'sharedUXPackages.userProfileComponents.userProfilesSelectable.searchPlaceholder',
+            {
+              defaultMessage: 'Search',
+            }
+          ),
+        onChange: onSearchChange,
+        isLoading,
+        isClearable: !isLoading,
+      }}
+      isPreFiltered
+      listProps={{ onFocusBadge: false }}
+      loadingMessage={loadingMessage}
+      noMatchesMessage={noMatchesMessage}
+      emptyMessage={emptyMessage}
+      errorMessage={errorMessage}
+    >
+      {(list, search) => (
+        <>
+          <EuiPanel hasShadow={false} paddingSize="s">
+            {search}
+            <EuiSpacer size="s" />
+            <EuiFlexGroup alignItems="center" justifyContent="spaceBetween" responsive={false}>
+              <EuiFlexItem grow={false}>
+                <EuiText size="xs" color="subdued">
+                  {selectedStatusMessage ? (
+                    selectedStatusMessage(selectedCount)
+                  ) : (
+                    <FormattedMessage
+                      id="sharedUXPackages.userProfileComponents.userProfilesSelectable.selectedStatusMessage"
+                      defaultMessage="{count, plural, one {# user selected} other {# users selected}}"
+                      values={{ count: selectedCount }}
+                    />
+                  )}
+                </EuiText>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                {selectedCount ? (
+                  <EuiButtonEmpty
+                    size="xs"
+                    flush="right"
+                    onClick={() => onChange?.([])}
+                    style={{ height: '1rem' }}
+                  >
+                    {clearButtonLabel ?? (
+                      <FormattedMessage
+                        id="sharedUXPackages.userProfileComponents.userProfilesSelectable.clearButtonLabel"
+                        defaultMessage="Remove all users"
+                      />
+                    )}
+                  </EuiButtonEmpty>
+                ) : undefined}
```

Comment:
nit: should be `null`

Reply:
Nice good catch [702fb2e](https://github.com/elastic/kibana/pull/139247/commits/702fb2e4725f60d47cc2989e440b98053b042a7f)